From fa2d0ca3c31fdf70643ff68611ea0f6ad8b12519 Mon Sep 17 00:00:00 2001
From: Christophe Kerello <christophe.kerello@foss.st.com>
Date: Wed, 3 Jul 2024 17:49:39 +0200
Subject: [PATCH 645/841] feat(stm32mp21-fdts): add eMMC support for
 stm32mp215f-dk board

Enable SDMMC2 node for stm32mp215f-dk board.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 704aaa53cf2fffbd7d16790f992d08afd4b8c3dd feat(stm32mp21-fdts): add eMMC support for stm32mp215f-dk board]
Change-Id: I1f9869c5b70ff98c89b5de6bfcd01f6eff00c6b4
Signed-off-by: Christophe Kerello <christophe.kerello@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/391264
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Tested-by: Christophe KERELLO <christophe.kerello@st.com>
Reviewed-by: Christophe KERELLO <christophe.kerello@st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Christophe KERELLO <christophe.kerello@st.com>
---
 fdts/stm32mp215f-dk.dts | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/fdts/stm32mp215f-dk.dts b/fdts/stm32mp215f-dk.dts
index 082218da0..f183a6d2a 100644
--- a/fdts/stm32mp215f-dk.dts
+++ b/fdts/stm32mp215f-dk.dts
@@ -188,6 +188,19 @@
 	status = "okay";
 };
 
+&sdmmc2 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&sdmmc2_b4_pins_a &sdmmc2_d47_pins_a>;
+	non-removable;
+	no-sd;
+	no-sdio;
+	st,neg-edge;
+	bus-width = <8>;
+	vmmc-supply = <&v3v3>;
+	vqmmc-supply = <&vddio2>;
+	status = "okay";
+};
+
 &usart2 {
 	pinctrl-names = "default", "idle", "sleep";
 	pinctrl-0 = <&usart2_pins_a>;
-- 
2.47.2

